dev/run-tests

```diff
@@ -17,239 +17,394 @@
 # limitations under the License.
 #
 
-# Go to the Spark project root directory
-FWDIR="$(cd "`dirname $0`"/..; pwd)"
-cd "$FWDIR"
+import os
+import re
+import shutil
+import subprocess as sp
```

Comment:
I suggest not renaming this module as the `sp.` abbreviation will be a bit confusing below. Better to just leave the calls as `subprocess.`. Clearer that way.